tests/consumer/test_oauth2.py

```diff
@@ -127,6 +129,24 @@ def test_authorized_url():
         )
 
 
+def test_authorized_url_invalid_response():
+    app, _ = make_app(debug=True)
+    with app.test_client() as client:
+        # reset the session before the request
+        with client.session_transaction() as sess:
+            sess["test-service_oauth_state"] = "random-string"
+        # make the request
+        with pytest.raises(MissingCodeError) as missingError:
+            client.get(
+                "/login/test-service/authorized?error_code=1349048&error_message=IMUSEFUL",
+                base_url="https://a.b.c",
+            )
+        assert str(missingError.value) == ('(missing_code) The redirect request did not contain a token. '
+                                           'Instead I got: {"error_message": "IMUSEFUL", "error_code": "1349048"}')
```

Comment:
Parsing the dict may be difficult.

ill poke at this some. I may do the approach of checking for key words.

For the purpose of the test I just wanna make sure each word is in there

```
assert all(keyword in missingError.value for keyword in ["error_message",  "IMUSEFUL", "error_code",  "1349048"]
```

but when I am with the IDE tonight ill look for other solutions